plugin/storage/es/options.go

```diff
@@ -157,30 +166,30 @@ func addFlags(flagSet *flag.FlagSet, nsConfig *namespaceConfig) {
 		nsConfig.namespace+suffixTimeout,
 		nsConfig.Timeout,
 		"Timeout used for queries. A Timeout of zero means no timeout")
-	flagSet.Int64(
+	flagSet.Int(
 		nsConfig.namespace+suffixNumShards,
-		nsConfig.NumShards,
+		defaultIndexOptions.Shards,
```

Comment:
This does not appear to be an equivalent change. Previously defaults were read from a struct passed as argument. Now they are read from a static struct. I think the intention of the original pattern was that you have a CreateDeafaultConfig function somewhere which populates stuff (including for v2) and cli flags don't invent their own defaults. Can we put it back to the original design?

Reply:
Since cli flags don't allow per-index overrides for many options we can read them from the spans index settings.